### PR TITLE
feat: implement computation of arb rewards

### PR DESCRIPTION
**Pull request by amateima**
This PR implements the computation of ARB rebate rewards for deposits to Arbitrum

### CODE REVIEW

File: migrations/1718123954440-ArbReward.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    await queryRunner.query(`
+      ALTER TABLE "arb_reward" 
+      ADD CONSTRAINT "FK_arb_reward_deposit" 
+      FOREIGN KEY ("depositPrimaryKey") REFERENCES "deposit"("id") ON DELETE NO ACTION ON UPDATE NO ACTION
+    `);
+    await queryRunner.query(`
+      ALTER TABLE "arb_reward" 
+      ADD CONSTRAINT "FK_arb_reward_token" 
+      FOREIGN KEY ("rewardTokenId") REFERENCES "token"("id") ON DELETE NO ACTION ON UPDATE NO ACTION
+    `);
```
Comment (amateima) [inline migrations/1718123954440-ArbReward.ts:35]: SQL migration queries for creating the table for arb rebate rewards

File: src/modules/configuration/index.ts [hunk 2]
```diff
@@ -316,6 +316,15 @@ export const configValues = () => ({
...
       },
+      arbRewards: {
+        chainId: process.env.ARB_REWARDS_MERKLE_DISTRIBUTOR_CHAIN_ID
+          ? Number(process.env.ARB_REWARDS_MERKLE_DISTRIBUTOR_CHAIN_ID)
+          : 42161,
+        address: process.env.ARB_REWARDS_MERKLE_DISTRIBUTOR_ADDRESS || "0x9f6Cb0A37F1ae91b8e65405f525A596bAFC5A9a6",
+        blockNumber: process.env.ARB_REWARDS_MERKLE_DISTRIBUTOR_BLOCK_NUMBER
+          ? Number(process.env.OP_REWARDS_MERKLE_DISTRIBUTOR_BLOCK_NUMBER)
+          : 220861533,
+      },
```
Comment (amateima) [inline src/modules/configuration/index.ts:327]: The config values for the MerkleDistributor contract used for distributing the referral rewards. The fallback values represent the contract deployed on mainnet

File: src/modules/rewards/services/arb-rebate-service.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  public async createArbRebatesForDeposit(depositPrimaryKey: number) {
+    if (!this.appConfig.values.rewardPrograms.arbRebates.enabled) {
+      this.logger.verbose(`ARB rebate rewards are disabled. Skipping...`);
+      return;
+    }
+
+    const deposit: PartialDeposit = await this.depositRepository.findOne({
+      where: { id: depositPrimaryKey },
+      select: partialDepositKeys,
+    });
```
Comment (amateima) [inline src/modules/rewards/services/arb-rebate-service.ts:54]: This is something new, but I incline to make it a practice from now on. Fetch only the columns we are interested in, not the entire row from the `deposit` table

File: src/modules/rewards/services/arb-rebate-service.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        recipient: rewardReceiver,
+        depositDate: deposit.depositDate,
+        metadata: {
+          rate: ARB_REBATE_RATE,
+        },
+        amount: rewardsAmount.toString(),
+        amountUsd: rewardsUsd,
+        rewardTokenId: rewardToken.id,
+      })
+      .orUpdate(["recipient", "depositDate", "metadata", "amount", "amountUsd", "rewardTokenId"], ["depositPrimaryKey"])
```
Comment (amateima) [inline src/modules/rewards/services/arb-rebate-service.ts:87]: if reward for deposit already exist, overwrite the values

File: src/modules/scraper/entry-point/http/controller.ts [hunk 5]
```diff
@@ -231,6 +232,21 @@ export class ScraperController {
...
+  async submitArbRebateRewardJob(@Body() body: OpRebateRewardBody) {
+    const { fromDepositId, toDepositId } = body;
+
+    for (let depositId = fromDepositId; depositId <= toDepositId; depositId++) {
+      this.scraperQueuesService.publishMessage<ArbRebateRewardMessage>(ScraperQueue.ArbRebateReward, {
+        depositPrimaryKey: depositId,
+      });
+    }
+  }
+
```
Comment (amateima) [inline src/modules/scraper/entry-point/http/controller.ts:249]: admin endpoint to be able to trigger quickly re-computation of arb rewards, if needed

**Review by melisaguevara — APPROVED** [review]
LGTM!

**Pull request merged**